charts/kong/values.yaml

```diff
@@ -518,24 +518,142 @@ dblessConfig:
 # created in k8s. It uses CRDs for more fine grained control over routing and
 # for Kong specific configuration.
 ingressController:
-  enabled: true
```

Comment:
As I agree with moving stuff strictly related to Deployment under `ingressController.deployment` (it avoids building unnecessary abstractions over vanilla K8s API), I'm not convinced we should move this toggle there though. It's not strictly related to the deployment only, but I sense it's more of a feature toggle from the Helm chart perspective. Rendering a controller Deployment is one of its effects, but it also determines whether to generate ServiceAccount,  RBACs, etc. 

Is there any advantage in moving it that I missed?

Reply:
Maybe more an artifact of the original design (#923) than anything, which had `deployment.kong.enabled` and `deploment.controller.enabled`. Given that this won't have like configuration paths anyway, there's probably no good reason to move it.